### PR TITLE
Support aggregate expression in  select statement without from clause

### LINKED ISSUE #4130: wrong nullable property when Implicit happen
### Steps to reproduce the behavior (Required)
```
select 'x' - 'x' union select 0
```

child 1 should be nullable column
```
 0:UNION
  |  child exprs:
  |      [1, DOUBLE, false]
  |      [2, DOUBLE, false]
  |  pass-through-operands: all
  |  cardinality: 2
  |  
  |----4:EXCHANGE
  |       cardinality: 1
  |    
  2:EXCHANGE
     cardinality: 1

PLAN FRAGMENT 3(F02)

  Input Partition: UNPARTITIONED
  OutPut Partition: HASH_PARTITIONED: <slot 2>
  OutPut Exchange Id: 04

  3:UNION
     constant exprs: 
         0.0
     cardinality: 1

PLAN FRAGMENT 4(F01)

  Input Partition: UNPARTITIONED
  OutPut Partition: HASH_PARTITIONED: <slot 1>
  OutPut Exchange Id: 02

  1:UNION
     constant exprs: 
         CAST('x' AS DOUBLE) - CAST('x' AS DOUBLE)
     cardinality: 1
```

### StarRocks version (Required)
 - main 2.1.1 2.0.2

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4130 #4124 

## Problem Summary(Required) ：

1. The syntax sql of from == null is changed from the value relation to the select relation
2. Moved part of UT to planFragmentTest

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 47 / 54 (87.04%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|5|12|41.67%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|28|28|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ValuesRelation.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 47 / 54 (87.04%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|5|12|41.67%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|28|28|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ValuesRelation.java|3|3|100.00%|

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Review by murphyatwork — DISMISSED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 67 / 69 (97.10%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SqlDigestBuilder.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|28|28|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ValuesRelation.java|3|3|100.00%|

**Review by Seaven — DISMISSED** [review]

**Review by liuyehcf — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java [hunk 1]
```diff
@@ -48,7 +48,11 @@ public ExpressionStatisticVisitor(Statistics statistics, double rowCount) {
 
         @Override
         public ColumnStatistic visit(ScalarOperator operator, Void context) {
-            return operator.getChild(0).accept(this, context);
+            if (operator.getChildren().size() > 1) {
+                return operator.getChild(0).accept(this, context);
+            } else {
+                return ColumnStatistic.unknown();
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java:54]: select [][1] will report an error

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/SetTest.java [hunk 2]
```diff
@@ -183,16 +183,24 @@ public void testSetOperation() throws Exception {
...
-        Assert.assertTrue(plan.contains("  10:UNION\n" +
+        Assert.assertTrue(plan.contains("  3:Project\n" +
+                "  |  <slot 2> : 499\n" +
+                "  |  \n" +
+                "  2:UNION\n" +
                 "     constant exprs: \n" +
-                "         499"));
-        Assert.assertTrue(plan.contains("1:UNION"));
-        Assert.assertTrue(plan.contains("  4:UNION\n" +
+                "         NULL"));
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/SetTest.java:191]: null is the null generated in values, the original 1 is located in the project, and this column in the project is pruned during optimization

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/PlanTestBase.java [hunk 3]
```diff
@@ -1120,4 +1121,13 @@ private void checkWithIgnoreTabletList(String expect, String actual) {
 
         Assert.assertEquals(expect, actual);
     }
+
+    protected void assertPlanContains(String sql, String... explain) throws Exception {
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/PlanTestBase.java:1125]: This is the original function, moved the position

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 67 / 69 (97.10%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SqlDigestBuilder.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|28|28|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ValuesRelation.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 67 / 69 (97.10%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SqlDigestBuilder.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|28|28|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ValuesRelation.java|3|3|100.00%|

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 90 / 102 (88.24%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SqlDigestBuilder.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SysVariableDesc.java|1|11|09.09%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|28|28|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ExpressionStatisticCalculator.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/analysis/FunctionCallExpr.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/ValuesRelation.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/CallOperator.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeState.java|2|2|100.00%|

**Review by Youngwb — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**